### PR TITLE
Takeaway Challenge: Russell Vaughan

### PR DESCRIPTION
**Pull request by russellvaughan**

### CODE REVIEW

File: spec/feature_spec.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  end
+
+
+  it "I would like to receive a text when order is placed" do 
+  takeaway = Takeaway.new
+  takeaway.select_item("Roast Pork Belly Hirata Buns")
+  takeaway.select_item("Salt & Pepper Squid",2)
+  takeaway.select_item("Roast Pork Belly Hirata Buns")
+  thank_you_message ="Thank you! Your order was placed and will be delivered before 20:00"
+  allow(takeaway).to receive(:one_hours_time) {"20:00"}
```
Comment (yyl29) [inline spec/feature_spec.rb]: good stubbing for time!

File: lib/takeaway.rb [hunk 2]
```diff
@@ -0,0 +1,93 @@
+require './lib/menu'
+require './lib/messaging'
+
+class Takeaway
+
+attr_reader :order, :bill
+
+	def initialize
+	@menu = Menu.new
```
Comment (yyl29) [inline lib/takeaway.rb]: menu and message could pass in as argument when initialise(dependency injection)

File: lib/takeaway.rb [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+def incorrect_total(amount)
+	amount != bill
+end
+
+def one_hours_time
+	time = Time.new + 60 * 60	
+	time.strftime("%H:%M")
+end
+
+def print_menu
```
Comment (yyl29) [inline lib/takeaway.rb]: like the menu!

File: lib/takeaway.rb [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+end
+
+def subtotal_of_items
+items = ""
+subtotal = 0
+duplicates = Hash.new(0)
+    @order.each do |dish|
+ 	duplicates[dish] += 1
+ 	end	
+ 	duplicates.each do |dish, quantity|
```
Comment (yyl29) [inline lib/takeaway.rb]: comes out very nicely!

**Comment by mechacoach** [comment]
You had your takeaway-challenge reviewed by **Yan-Yi Li**

### The good points are:

* The README has been updated
* The README has instructions
* Required gems (i.e. twilio-ruby) are listed in Gemfile
* All tests test real behaviour and not stubs
* The Twilio API is stubbed appropriately
* All unit tests use doubles (London style)
* All public interface elements are tested
* The solution correctly prefers classes to modules
* The solution makes good use of the Ruby language, e.g., Hash.new(0) is used to simplify management of counts
* The open closed principle has been respected, e.g. food items are not hard coded into restaurant
* The code follows the Ruby style and coding conventions

### You should consider the following improvements:

* Use named subject with `described_class`
* Law of Demeter
* Appropriate use of Dependency Injection
* Separation of Concerns
* Design for Single Responsibility Principle
* Personal details and tokens on GitHub

### Additional comments:
really like the layout of menu and summary, like how message is in different class. could use dependency injection for message and menu.

see https://github.com/makersacademy/takeaway-challenge/blob/master/docs/review.md for more details

**Pull request closed without merging**